src/components/Banner/Banner.tsx

```diff
@@ -1,37 +1,72 @@
-import React, { useState, Dispatch, SetStateAction } from 'react';
+import React from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+import {
+  faCar,
+  faMoneyBillAlt,
+  faCreditCard,
+  IconDefinition,
+} from '@fortawesome/free-solid-svg-icons';
 import './Banner.css';
+import { IDiscountWithNote } from '../../types/types';
 
 interface IProps {
-  show: boolean;
-  setShow: Dispatch<SetStateAction<boolean>>;
-  message?: string;
-  tooltipMessages?: string[];
+  className?: string;
+  delivery?: string;
+  cash?: string | IDiscountWithNote;
+  card?: string | IDiscountWithNote;
 }
 
 const Banner = (props: IProps) => {
-  const [tooltipHidden, setTooltipHidden] = useState(true);
-  const { show, setShow, message, tooltipMessages } = props;
+  const { className, delivery, cash, card } = props;
 
-  const toggleTooltip = () => setTooltipHidden(!tooltipHidden);
-
-  return message ? (
-    <div className={`banner ${!show && 'dismissed'}`}>
-      <div onMouseEnter={toggleTooltip} onMouseLeave={toggleTooltip}>
-        <FontAwesomeIcon icon={faInfoCircle} />
-        <div className={`tooltip ${tooltipHidden && 'hidden'}`}>
-          {tooltipMessages?.map((message, index) => (
-            <li key={index}>{message}</li>
-          ))}
-        </div>
-      </div>
-      <div className="message">{message}</div>
-      <div className="close" onClick={() => setShow(false)}>
-        ✕
-      </div>
+  return (
+    <div className={`banner ${className}`}>
+      <BannerItem description={delivery} icon={faCar} />
+      <BannerItem
+        description={
+          typeof cash === 'string'
+            ? cash
```

Comment:
This part will probably have to just check if note is null if you decide to do the changes